### ISSUE TITLE
"move window one monitor to the right/left" doesn't work with tiling enabled

### ISSUE THREAD

**Issue by carlwgeorge** [pop-os/shell#522]
**Distribution (run `cat /etc/os-release`):**

```
NAME=Fedora
VERSION="32 (Workstation Edition)"
ID=fedora
VERSION_ID=32
VERSION_CODENAME=""
PLATFORM_ID="platform:f32"
PRETTY_NAME="Fedora 32 (Workstation Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:32"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f32/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=32
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=32
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```
 
**Related Application and/or Package Version (run `apt policy pop-shell` or provide the latest commit if building locally):**

gnome-shell-extension-pop-shell-0.1.0-0.2.20200821git8416328.fc32.noarch (built from commit 8416328 in [COPR](https://copr.fedorainfracloud.org/coprs/carlwgeorge/gnome-shell-extension-pop-shell/))

**Issue/Bug Description:**

Moving a window one monitor to the left or right via the keyboard shortcut does not work with tiling enabled.  The window does initially move as instructed, but immediately returns to the monitor it started on.  This is reproducible in a Pop!_OS live environment as well, so I don't believe it's a problem with the RPM packaging.  Here is a video of the behavior.

https://imgur.com/RpuI3xp

**Steps to reproduce (if you know):**

1. enable tiling mode
2. create one or more windows on the first monitor
3. press `<Shift><Super>Right` to move a window to the second monitor

**Expected behavior:**

window moves to second monitor and stays there

**Comment by jmmaranan** [comment]
@carlwgeorge - can you please go to Gnome Tweaks > Workspaces > Display Handling, and see if `Workspaces span displays` is enabled?

**Comment by carlwgeorge** [comment]
Yes, I have have `workspaces-only-on-primary` set to false, just like the [Pop!_OS default setting](https://github.com/pop-os/session/blob/stacking_focal/debian/pop-session.gsettings-override#L92).  Setting that to true allows me to move a window to the right or left monitor, but then I can't move it back to the middle (primary) monitor.

**Comment by carlwgeorge** [comment]
I was able to reproduce this with both two and three monitor setups.

**Comment by jmmaranan** [comment]
> Yes, I have have `workspaces-only-on-primary` set to false, just like the [Pop!_OS default setting](https://github.com/pop-os/session/blob/stacking_focal/debian/pop-session.gsettings-override#L92). Setting that to true allows me to move a window to the right or left monitor, but then I can't move it back to the middle (primary) monitor.

@carlwgeorge - I think you meant `false`? How many monitors do you have? Laptop with two external monitors?

@jacobgkau - I was just wondering if the Github issue template can be updated to explicitly ask the monitor setup? Or other details.

**Comment by jacobgkau** [comment]
> @jacobgkau - I was just wondering if the Github issue template can be updated to explicitly ask the monitor setup? Or other details.

Yes, that's possible. The issue template is located here: https://github.com/pop-os/shell/blob/master_focal/.github/ISSUE_TEMPLATE.md 

> @carlwgeorge - I think you meant `false`? How many monitors do you have? Laptop with two external monitors?

Like @carlwgeorge said, the default setting for `workspaces-only-on-primary` in Pop!_OS is `false`, which corresponds to the `Workspaces span displays` option in GNOME Tweaks. With the default setting and two displays, I am not able to switch windows between the displays with Super+Shift+Left/Right at all.

If I change the GNOME Tweaks setting to `Workspaces on primary display only` (which sets `workspaces-only-on-primary` to `true`), then I'm able to use Super+Shift+arrow to move windows from the primary display (left) to the secondary display (right), but I'm not able to move windows from the secondary display to the primary display.

When I add a third display (with the layout being Primary, Secondary, Secondary), with the non-default `Workspaces on primary display only`, I'm able to move some of my windows back and forth between the leftmost and center displays (but not onto the rightmost display), while other windows won't move at all. With the Pop default setting of `Workspaces span displays`, none of the windows ever move between any of the three displays with these keyboard shortcuts. I hope this helps clarify the issue.

**Comment by carlwgeorge** [comment]
@jmmaranan

> I think you meant false?

No I meant exactly what I said.  That setting was false when I first reported this issue, and per your request in https://github.com/pop-os/shell/issues/522#issuecomment-683329020 I changed it to true and then had the behavior described in https://github.com/pop-os/shell/issues/522#issuecomment-683359872.

> How many monitors do you have? Laptop with two external monitors?

Not a laptop, a custom PC with a [Sapphire RX 580](https://www.sapphiretech.com/en/consumer/pulse-rx-580-8g-g5) video card driving three [Dell U2718Q](https://www.dell.com/support/home/en-us/product-support/product/dell-u2718q-monitor/overview) monitors.

@jacobgkau Thanks for confirming that you can reproduce the behavior.  For reference my monitor layout is secondary-primary-secondary.

**Comment by bjonnh** [comment]
I have a dual screen setup (two external screens on a laptop with its internal disabled).

I confirm that changing the option in gnome tweaks to "workspaces on primary only" solve the issue. But of course now I only have workspaces on my primary…

**Comment by TC68** [comment]
Same issue. Laptop (Gazelle) with one external monitor connected via USBC. If is have "workspaces span displays" activated, my work spaces allow windows on both monitors but I can't use keyboard shortcuts or the window menu to successfully move windows to the other monitor. Moving up and down workspaces works fine.

If I enable "Workspaces on primary monitor only", I can't use the external monitor as effectively and I still have problems moving the window. I can "Move window to monitor right" successfully. This moves my window from the primary monitor to the external monitor. I can not use the shortcuts to move the window back from the external monitor to the primary display, either using keyboard shortcut or window menu. Only mouse dragging will do it.

**Comment by lorenzo-arena** [comment]
Should this be fixed? As in Pop 21.04 with tiling enabled while I can move windows from primary to secondary display using <Super><Shift><Right> I can not move windows back to the primary display with <Super><Shift><Left>. I have "Workspaces span displays" enabled in Gnome Tweaks.

**Comment by murar8** [comment]
Same issue on Fedora 40 using gnome-shell-extension-pop-shell-1.2.0^22.5fa3a50-1.fc40.noarch

EDIT: For me the issue was that the keybinding was executing `org/gnome/desktop/wm/keybindings/move-to-monitor-X` instead of `org/gnome/shell/extensions/pop-shell/pop-monitor-X`
